tsl/src/bgw_policy/job_api.c

```diff
@@ -235,6 +304,52 @@ job_alter(PG_FUNCTION_ARGS)
 	if (!PG_ARGISNULL(6))
 		job->fd.config = PG_GETARG_JSONB_P(6);
 
+	if (!PG_ARGISNULL(9))
+	{
+		if (OidIsValid(check))
+		{
+			check_name_str = get_func_name(check);
+			if (check_name_str == NULL)
+				ereport(ERROR,
+						(errcode(ERRCODE_UNDEFINED_OBJECT),
+						 errmsg("function with OID %d does not exist", check)));
```

Comment:
Perhaps you can convert this in an assertion, in the API you're accepting a regproc so just valid oids from pg_proc are accepted or nulls (aka InvalidOid).

Reply:
regproc does not mean it's a valid oid so the check is necessary
```
sven@dev[2194025]=# SELECT 123456789::regproc;
  regproc
-----------
 123456789
(1 row)
```

Reply:
thanks @svenklemm ... so @konskov we need to add test for it

Reply:
Thank you - done